apps/emqx/src/emqx_frame.erl

```diff
@@ -182,8 +182,19 @@ parse_remaining_len(
     Packet = packet(Header, #mqtt_packet_disconnect{reason_code = ?RC_SUCCESS}),
     {ok, Packet, Rest, ?NONE(Options)};
 %% Match PINGREQ.
-parse_remaining_len(<<0:8, Rest/binary>>, Header, 1, 0, Options) ->
+parse_remaining_len(
+    <<0:8, Rest/binary>>, Header = #mqtt_packet_header{type = ?PINGREQ}, 1, 0, Options
+) ->
+    parse_frame(Rest, Header, 0, Options);
+parse_remaining_len(
+    <<0:8, Rest/binary>>, Header = #mqtt_packet_header{type = ?PINGRESP}, 1, 0, Options
+) ->
     parse_frame(Rest, Header, 0, Options);
+%% Without this clause a crash may will happen when data incorrect, this was found by fuzzing test
+parse_remaining_len(
+    <<0:8, _Rest/binary>>, _Header, 1, 0, _Options
+) ->
+    ?PARSE_ERR(invalid_remaining_len);
```

Comment:
maybe raise a map with `#{hint => expecting_non_empty_payload, header_type => Header#mqtt_packet_header.type}` here so we know which